### PR TITLE
GaugeController: Add Events & Natspec

### PR DESCRIPTION
**Pull request by kamescg**

### CODE REVIEW

File: contracts/GaugeController.sol [hunk 1]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
 
     /**
-     * @notice Event emitted when the contract is deployed
-     * @param token Address of the token being staked in the gauge
+      * @notice Emitted when User deposits 'token' into the gauge. 
+      * @param user Address of the user who deposited 'token' into the GaugeController
+      * @param amount Amount of 'token' deposited into the GaugeController
      */
-     event Deployed(IERC20 token, IGaugeReward gaugeReward, address owner);
+     event UserDeposit(address user, uint256 amount);
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply contracts/GaugeController.sol:85]: The `User` and `Authority` pattern is pretty interesting but it feels a bit redundant.
If needed, we should be able to retrieve the `user` address from the event and then compare it to the `owner` and `manager` if we wished to know if this event was emitted in a role restricted function.

[the remark this answers is not in the document]
Comment (PierrickGT) [reply contracts/GaugeController.sol:85]: Also, I think we should follow the OpenZeppelin naming convention and name events in past tense.
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/GUIDELINES.md#solidity-code

[the remark this answers is not in the document]
Comment (PierrickGT) [reply contracts/GaugeController.sol:85]:
```suggestion
     event TokenDeposited(address indexed user, IERC20 indexed token, uint256 amount);
```

File: contracts/GaugeController.sol [hunk 2]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+     
+     /**
+      * @notice Emitted when User withdraws 'token' from the gauge. 
+      * @param user Address of the user who withdrew 'token' from the GaugeController
+      * @param amount Amount of 'token' withdrawn from the GaugeController
+    */
+    event UserWithdraw(address user, uint256 amount);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:92]:
```suggestion
    event TokenWithdrawn(address indexed user, IERC20 indexed token, uint256 amount);
```

File: contracts/GaugeController.sol [hunk 3]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+    
+    /**
+      * @notice Emitted when User increases a Gauge weight.
+      * @param user User address
+      * @param gauge Existing Gauge address
+      * @param amount Amount of 'balance' debited from the User and credited to the Gauge
+    */
+      event UserGaugeIncrease(address user, address gauge, uint256 amount);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:100]:
```suggestion
      event GaugeIncreased(address indexed user, address indexed gauge, uint256 amount);
```

File: contracts/GaugeController.sol [hunk 4]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+    
+    /**
+      * @notice Emitted when User decrease a Gauge weight.
+      * @param user User address
+      * @param gauge Existing Gauge address
+      * @param amount Amount of 'balance' debited from the Gauge and credited to the User
+    */
+    event UserGaugeDecrease(address user, address gauge, uint256 amount);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:108]:
```suggestion
    event GaugeDecreased(address indexed user, address indexed gauge, uint256 amount);
```

File: contracts/GaugeController.sol [hunk 5]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+
+    /**
+     * @notice Emitted when an Authorized User adds a new Gauge to the GaugeController
+     * @param gauge New Gauge address
+    */
+    event AuthorityAddGauge(address gauge);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:114]: We should also pass the `scale`:
```suggestion
    event GaugeAdded(address indexed user, address indexed gauge, uint256 scale);
```

File: contracts/GaugeController.sol [hunk 6]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+    
+    /**
+     * @notice Emitted when an Authorized User removes an existing Gauge from the GaugeController
+     * @param gauge New Gauge address
+    */
+    event AuthorityRemoveGauge(address gauge);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:120]:
```suggestion
    event GaugeRemoved(address indexed user, address indexed gauge);
```

File: contracts/GaugeController.sol [hunk 7]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+
+    /**
+     * @notice Emitted when an Authorized User sets an existing Gauge 'scale' weight.
+     * @param gauge New Gauge address
+     * @param scale New Gauge scale
+     * @param oldScale Old Gauge scale
+    */
+    event AuthoritySetGaugeScale(address gauge, uint256 scale, uint256 oldScale);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:128]:
```suggestion
    event GaugeScaleSet(address indexed user, address indexed gauge, uint256 scale, uint256 oldScale);
```

File: contracts/GaugeController.sol [hunk 8]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+
+    /**
+     * @notice Emitted when an Authorized User sets an existing Gauge 'reward' weight.
+     * @param gaugeReward New GaugeReward address
+     * @param oldGaugeReward Old GaugeReward address
+    */
+    event AuthoritySetGaugeReward(address gaugeReward, address oldGaugeReward);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:135]:
```suggestion
    event GaugeRewardSet(address indexed user, address indexed gaugeReward, address indexed oldGaugeReward);
```

File: contracts/GaugeController.sol [hunk 9]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+    
+    /**
+     * @notice Emitted when an Authorized User removes an existing Gauge from the GaugeController
+     * @param gauge New Gauge address
+    */
+    event AuthorityRemoveGauge(address gauge);
+
+    /**
+     * @notice Emitted when an Authorized User sets an existing Gauge 'scale' weight.
+     * @param gauge New Gauge address
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:124]:
```suggestion
     * @param gauge Gauge address
```

File: contracts/GaugeController.sol [hunk 10]
```diff
@@ -65,17 +65,74 @@ contract GaugeController is IGaugeController, Ownable {
...
+
+    /**
+     * @notice Emitted when an Authorized User adds a new Gauge to the GaugeController
+     * @param gauge New Gauge address
+    */
+    event AuthorityAddGauge(address gauge);
+    
+    /**
+     * @notice Emitted when an Authorized User removes an existing Gauge from the GaugeController
+     * @param gauge New Gauge address
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:118]:
```suggestion
     * @param gauge Gauge address
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: contracts/GaugeController.sol [hunk 11]
```diff
@@ -215,7 +273,6 @@ contract GaugeController is IGaugeController, Ownable {
      function setGaugeReward(IGaugeReward _gaugeReward) external onlyOwner {
         require(address(_gaugeReward) != address(0), "GC/GaugeReward-not-zero-address");
         gaugeReward = _gaugeReward;
-
         emit GaugeRewardSet(_gaugeReward);
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:276]: We could pass the `msg.sender` here.

File: contracts/GaugeController.sol [hunk 12]
```diff
@@ -66,16 +66,68 @@ contract GaugeController is IGaugeController, Ownable {
...
+    //  * @notice Emitted when GaugeReward address is set/updated
+    //  * @param gaugeReward Address of the newly set GaugeReward contract
+    // */
     event GaugeRewardSet(IGaugeReward gaugeReward);
 
     /**
-     * @notice Event emitted when the contract is deployed
-     * @param token Address of the token being staked in the gauge
+      * @notice Emitted when User deposits 'token' into the gauge. 
+      * @param user Address of the user who deposited 'token' into the GaugeController
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:82]: Add Natspec doc for `token`.

File: contracts/GaugeController.sol [hunk 13]
```diff
@@ -66,16 +66,68 @@ contract GaugeController is IGaugeController, Ownable {
...
+      * @param amount Amount of 'token' deposited into the GaugeController
      */
-     event Deployed(IERC20 token, IGaugeReward gaugeReward, address owner);
+     event TokenDeposited(address indexed user, IERC20 indexed token, uint256 amount);
+     
+     /**
+      * @notice Emitted when User withdraws 'token' from the gauge. 
+      * @param user Address of the user who withdrew 'token' from the GaugeController
```
Comment (PierrickGT) [inline contracts/GaugeController.sol:89]: Add Natspec doc for `token`.

File: contracts/GaugeController.sol (hunk 5 above)
Comment (PierrickGT) [reply contracts/GaugeController.sol:114]: Need to be renamed `GaugeAdded`.

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: contracts/GaugeController.sol (hunk 11 above)
Comment (kamescg) [reply contracts/GaugeController.sol:276]: It's the owner, so didn't really see a need. Maybe if it's possible from multiple accounts in the future?

**Pull request merged**